text/case_test.ts

```diff
@@ -0,0 +1,246 @@
+});
+
+Deno.test({
+  name: "toScreamingSnake() trims whitespace",
+  fn() {
+    const result = toScreamingSnakeCase(" deno Is AWESOME ");
+    const expected = "DENO_IS_AWESOME";
+    assertEquals(result, expected);
+  },
+});
+
+Deno.test({
+  name: "toScreamingSnakeCase() converts multiple delimiters",
+  fn() {
+    const result = toScreamingSnakeCase("I am up-to-date!");
+    const expected = "I_AM_UP_TO_DATE";
+    assertEquals(result, expected);
+  },
+});
+
+Deno.test("toSentenceCase() handles an empty string", () => {
+  assertEquals(toSentenceCase(""), "");
+});
+
+Deno.test("toSentenceCase() converts a single word", () => {
+  const input = "Shruberry";
+  const expected = "Shruberry";
+  assertEquals(toSentenceCase(input), expected);
+});
+
+Deno.test("toSentenceCase() converts a sentence", () => {
+  const input = "she turned me into a newt";
+  const expected = "She turned me into a newt";
+  assertEquals(toSentenceCase(input), expected);
+});
+
+Deno.test({
+  name: "toSentenceCase() converts multiple delimiters",
+  fn() {
+    const result = toSentenceCase("I am up-to-date!");
+    const expected = "I am up-to-date!";
+    assertEquals(result, expected);
+  },
+});
+
+Deno.test({
+  name: "toSentenceCase() trims whitespace",
+  fn() {
+    const result = toSentenceCase(" deno Is Awesome ");
+    const expected = "Deno is awesome";
+    assertEquals(result, expected);
+  },
+});
+
+Deno.test("toSnakeCase() handles an empty string", () => {
+  assertEquals(toSnakeCase(""), "");
+});
+
+Deno.test("toSnakeCase() converts a single word", () => {
+  const input = "shruberry";
+  const expected = "shruberry";
+  assertEquals(toSnakeCase(input), expected);
+});
+
+Deno.test("toSnakeCase() converts a sentence", () => {
+  const input = "she turned me into a newt";
+  const expected = "she_turned_me_into_a_newt";
+  assertEquals(toSnakeCase(input), expected);
+});
+
+Deno.test({
+  name: "toSnakeCase() converts multiple delimiters",
+  fn() {
+    const result = toSnakeCase("I am up-to-date!");
+    const expected = "i_am_up_to_date";
+    assertEquals(result, expected);
+  },
+});
+
+Deno.test({
+  name: "toSnakeCase() trims whitespace",
+  fn() {
+    const result = toSnakeCase(" deno Is AWESOME ");
+    const expected = "deno_is_awesome";
+    assertEquals(result, expected);
+  },
+});
+Deno.test("toTitleCase() handles an empty string", () => {
+  assertEquals(toTitleCase(""), "");
+});
+
+Deno.test("toTitleCase() converts a single word", () => {
+  const input = "shruberry";
+  const expected = "Shruberry";
+  assertEquals(toTitleCase(input), expected);
+});
+
+Deno.test("toTitleCase() converts a sentence", () => {
+  const input = "she turned me into a newt";
+  const expected = "She Turned Me Into A Newt";
+  assertEquals(toTitleCase(input), expected);
+});
+
+Deno.test({
+  name: "toTitleCase() converts multiple delimiters",
+  fn() {
+    const result = toTitleCase("I am up-to-date!");
+    const expected = "I Am Up-to-date!";
```

Comment:
Does this need to be like this?

Reply:
Well, it should be `I Am Up-to-Date!`, but title case is not easy to do correctly without analyzing language and grammar. As pointed out on https://titlecaseconverter.com:
> I	is capitalized because it is the first word of the title.
Am	is capitalized because it is a verb, and all verbs are capitalized in title case. More information on [capitalizing am](https://titlecaseconverter.com/blog/is-am-capitalized/)
Up	is capitalized because it seems to be part of an adjective or noun. More information on [capitalizing up](https://titlecaseconverter.com/blog/is-up-capitalized/) • [capitalizing hyphenated words](https://titlecaseconverter.com/blog/capitalizing-hyphenated-words/)
to	is not capitalized because it is used as a preposition. More information on [capitalizing to](https://titlecaseconverter.com/blog/is-to-capitalized/) • [capitalizing hyphenated words](https://titlecaseconverter.com/blog/capitalizing-hyphenated-words/)
Date	is capitalized because it is the last word of the title or subtitle. More information on [capitalizing hyphenated words](https://titlecaseconverter.com/blog/capitalizing-hyphenated-words/)